cg-spec/editors_draft.html

```diff
@@ -3031,11 +3033,101 @@ <h2>Alternatives to referential opacity</h2>
+        <h2>Selective referential transparency</h2>
+
+        <p>Since the need for referential opacity/transparency is use-case dependent (as discussed <a href="#ref-opacity">above</a>), it may make sense for different predicates to have different behaviors. Consider the following example.</p>
+
+        <pre id="selective-ref-opacity"
+        data-transform="updateExample"
+        data-content-type="text/turtle"
+        class="nohighlight example"
+        >
+        <!--
+        << dbr:Linköping dbo:populationTotal "104232"^^xsd::nonNegativeInteger >>
+          :at "2010-12-31"^^xsd:date ;
+          :source <https://dbpedia.org/data/Linköping.ttl> .
+
+        ### Candidate entailment #1
+
+        << dbr:Linköping dbo:populationTotal "0104232"^^xsd::nonNegativeInteger >>
+          :at "2010-12-31"^^xsd:date .
+
+        ### Candidate entailment #2
+
+        << dbr:Linköping dbo:populationTotal "0104232"^^xsd::nonNegativeInteger >>
+          :source <https://dbpedia.org/data/Linköping.ttl> .
+        -->
+        </pre>
+
+        <p>One could argue that the candidate entailment #1 is desirable because the property `:at` is understood to be about the <em>statement</em> made by the subject triple (that statement was true at the given date, regardless of its RDF expression). On the other hand, the candidate entailment #2 may be considered not desirable because the predicate `:source` is about the triple itself (that specific triple was parsed from the given Turtle file).</p>
 
-    <div class=note>Other design choices for the semantics have also been discussed by the community group; these have been <a href="https://github.com/w3c/rdf-star/issues/95">summarized in a Github issue</a>.</div>
+        <p>Subclasses of `rdf:Property` could therefore be defined to identify properties that imply referential transparency on their subject triple, on their object triple, or on both. Note however, that the group has not reached consensus on the definition or usefulness of such a vocabulary.</p>
+
+        <p>Here is a sketch of how to extend the semantics in a way that it can work with such properties. The basis of the idea is that each such property&nbsp;|p| is identified by adding to the RDF-star graph a triple of the form (|p|, `rdf:type`, `rdf-star:TransparencyEnforcingProperty`); e.g., for the previous example, this triple would be (`:at`, `rdf:type`, `rdf-star:TransparencyEnforcingProperty`). We call such a property a <dfn data-lt="TEP">transparency-enforcing property (TEP)</dfn>. The semantics of such TEPs can be captured for the different <a data-cite="RDF11-MT#dfn-entailment-regime">entailment regimes</a> by extending the corresponding notion of <a data-cite="RDF11-MT#dfn-interpretation">interpretation</a> as exemplified in the following for the <a data-cite="RDF11-MT#simple-interpretations">simple interpretation</a> of the <a data-cite="RDF11-MT#simpleentailment">simple entailment regime</a>.</p>
```

Comment:
```suggestion
        <p>Here is a sketch of how to extend the semantics in a way that it can work with such a property, which we call a <dfn data-lt="TEP">transparency-enforcing property (TEP)</dfn>. The basis of the idea is that each such property&nbsp;|p| is identified by adding to the RDF-star graph a triple of the form (|p|, `rdf:type`, `rdf-star:TransparencyEnforcingProperty`); i.e., for the previous example, this triple would be (`:at`, `rdf:type`, `rdf-star:TransparencyEnforcingProperty`). The semantics of such TEPs can be captured for the different <a data-cite="RDF11-MT#dfn-entailment-regime">entailment regimes</a> by extending the corresponding notion of <a data-cite="RDF11-MT#dfn-interpretation">interpretation</a> as exemplified in the following for the <a data-cite="RDF11-MT#simple-interpretations">simple interpretation</a> of the <a data-cite="RDF11-MT#simpleentailment">simple entailment regime</a>.</p>
```